### PR TITLE
Fix bugs with new actions setup.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Fix a few new CI bugs introduced by #2312:
- Instead of catching mypy errors present in this branch but not master, do it based on the fork point. This seems more intuitive and ignores many of the errors you see in #2329.
- In some cases docstrings diffs from autoformat were outputted different than I expected. This caused a flake8 crash. This remedies that.
- Missed copyright header in one file. A little disconcerting but such is.

**Testing steps**
Local testing.

### CODE REVIEW

**Comment by stephenroller** [comment]
image

Okay got mypy to do warnings instead.

**Comment by stephenroller** [comment]
This now requires an upstream merge in pytorch/add-annotations-github-action#3.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -81,17 +81,19 @@ jobs:
...
           mypy | sort | grep -v '^Found ' > typeerrors-HEAD.txt
-          git checkout -f master
+          forkpoint=$(git merge-base master HEAD)
+          git checkout -f $forkpoint
           mypy | sort | grep -v '^Found ' > typeerrors-MASTER.txt
           diff -u typeerrors-MASTER.txt typeerrors-HEAD.txt | grep -v '^+++' | grep '^+' > typeerrors-CHANGED.txt || true
           cat typeerrors-CHANGED.txt | sed 's/^+//' | sed 's/: error:/:1: mypy:/' | tee ${GITHUB_WORKSPACE}/output-annotations.txt
       - name: Add annotations
-        uses: pytorch/add-annotations-github-action@master
+        uses: stephenroller/add-annotations-github-action@master
```
Comment (EricMichaelSmith) [inline .github/workflows/lint.yml]: Just checked out your forked version - adding in a warning level is really useful! Will you see if you can get it merged into PyTorch?

Comment (stephenroller) [reply .github/workflows/lint.yml]: I'd like that, waiting on a review from the upstream maintainer.

**Pull request merged**